### PR TITLE
add files Categorycontroller categorySevice CategoryServiceI and method

### PR DESCRIPTION
**Pull request by DanteIturri**
add files for Category method delete
add method deleted in file CategoryController
add method complementary handleCategoryforid in file CategoryServiceImpl

### CODE REVIEW

File: src/main/java/com/alkemy/ong/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.alkemy.ong.service.CategoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/Category")
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/controller/CategoryController.java:13]: I think the routes should start lower case, this way /category?

Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/controller/CategoryController.java:13]: you are right the category is in lowercase

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.alkemy.ong.service.CategoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    @Autowired
+    CategoryEntityRepository categoryRepo;
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:14]: Maybe just CategoryRepository is stead of Category-Entity-repository

File: src/main/java/com/alkemy/ong/controller/CategoryController.java (hunk 1 above)
Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/controller/CategoryController.java:13]: now i make the change

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    public void deletedCategoryForId(String id) throws Exception {
+        CategoryEntity entity = this.handleFindById(id);
+        categoryRepo.delete(entity);
+    }
+    /*Method for Exist Category */
+    public CategoryEntity handleFindById(String id) throws Exception {
+        Optional<CategoryEntity> NoFoundCategory = categoryRepo.findById(id);
+        if (!NoFoundCategory.isPresent()) {
+            throw new Exception("The category no exist:"+id);
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:25]: "The category does not exits" is my suggestion here

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:14]: it is not with that name in source

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java (hunk 3 above)
Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java]: change message now, thank

**Comment by DanteIturri** [comment]
Apply change for comments, thank team!!

**Comment by JonathanBolatti** [comment]
this PR has conflicts, please resolve to approve ticket

**Comment by DanteIturri** [comment]
Ya resolví los conflictos

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (JonathanBolatti) [reply src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:14]: please, continue with the naming of names that we have been working on, for example: class name + container package

File: src/main/java/com/alkemy/ong/controller/CategoryController.java (hunk 1 above)
Comment (anavduo) [reply src/main/java/com/alkemy/ong/controller/CategoryController.java]: Maybe just /categories without the first /category, I think in the tickets only mentions categories, simpler

Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/controller/CategoryController.java]: hecho en la resolución de conflicto

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (DanteIturri) [reply src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:14]: hecho en la resolucion de conflicto

**Pull request merged**